src/contact/parallel/bvh_contact_manager.cc

```diff
@@ -16,16 +16,16 @@ namespace nimble {
       auto res = bvh::typed_narrowphase_result< NarrowphaseResult >();
       auto tree = build_snapshot_tree_top_down( _a.elements );
 
-      for ( auto &&elb : _b.elements )
-        query_tree_local( tree, elb, [&_a, &_b, &elb, &res, this]( std::size_t _i ) {
+      for ( auto &&elb : _b.elements ) {
+        query_tree_local(tree, elb, [&_a, &_b, &elb, &res, this](std::size_t _i) {
           const auto &face = _a.elements[_i];
           const auto &node = elb;
           ContactManager::PROJECTION_TYPE proj_type;
           ContactEntity::vertex closest_point;
           double gap;
           double normal[3];
```

Comment:
Do you need to store the unit normal vector for computing the contact force?

Reply:
Yes. The gap and normal are better/more convenient for commuting the contact force than just the projection point on the facet and the contacting node